annotations/annotators.py

```diff
@@ -152,22 +162,44 @@ def render_display(self, context={}):
         return render(self.context.get('request'), self.display_template, context)
 
     def get_context(self):
-        resource = self.get_resource()
+        """
+        Override to provide context for a specific word to annotate.
+        """
+        resource = self.get_resource()  # Get the tokenized content, which is a string
         request = self.context.get('request')
-        content = self.get_content(resource)
-        detect  = chardet.detect(content)
-        return {
+        
+        # Extract the word_id from the request
+        word_id = request.GET.get('word_id')
+        
+        # Find the word with the specified ID in the tokenized content
+        if word_id:
+            # Using regex to find the specific word with the given word_id
+            import re
+            word_pattern = re.compile(r'<word id="' + re.escape(word_id) + r'">(.*?)<\/word>')
+            match = word_pattern.search(resource)
+            if match:
+                specific_word = match.group(1)  # Extract the word text
+            else:
+                specific_word = None
+        else:
+            specific_word = None
```

Comment:
just define this before `if word_id` and then you don't need the two else statements above